packages/langium/src/serializer/json-serializer.ts

```diff
@@ -30,6 +32,13 @@ export interface AstNodeWithTextRegion extends AstNode {
     $textRegion?: AstNodeRegionWithAssignments;
 }
 
+/**
+ * {@link AstNode}s that may carry a semantically relevant comment.
+ */
+export interface AstNodeWithComment extends AstNode {
```

Comment:
Why is this defined in this file? Why not an optional property of `AstNode`?

Reply:
For the same reason that the `$sourceText` and `$textRegion` properties are defined in this file as well: They are only relevant on serialized AST nodes, and will not be set any AST node during normal operation of Langium.